src/loaders/lottie/tvgLottieModel.cpp

```diff
@@ -31,7 +31,33 @@
 /* Internal Class Implementation                                        */
 /************************************************************************/
 
+static inline float _applySmoothness(float value, float smoothness)
+{
+    if (smoothness >= 100.0f) return value;
+
+    smoothness /= 100.0f;
+    value = (value - (1.0f - smoothness) * 0.5f) / smoothness;
+    clamp(value, 0.0f, 1.0f);
+
+    return value;
+}
+
+
+static inline float _applyEasing(LottieInterpolator* interpolator, float value, float minEase, float maxEase)
+{
+    if (minEase == 0.0f && maxEase == 0.0f) return value;
 
+    Point in{0.0f, 0.0f};
+    Point out{1.0f, 1.0f};
+
+    if (minEase > 0.0f) in.x = minEase * 0.01f;
+    else in.y = -minEase * 0.01f;
+    if (maxEase > 0.0f) out.x = 1.0 - maxEase * 0.01f;
+    else out.y = 1.0 + minEase * 0.01f;
+
+    interpolator->set("", in, out);
```

Comment:
caching interpolator doesn't make sense, since there are too many of them. so the only idea I have now, is to introduce inside the interpolator struct another progress(value, in, out) fun, so that the samples[] values are not precalculated, since they will be used only once.